MetaMask/metamask-extension
# fix: Gracefully handle bad responses from `net_version` calls to RPC endpoint when getting Provider Network State

## **Description**

Currently some users are seeing `Error: Cannot parse as a valid network ID: 'undefined'` errors when connected to RPC endpoints that fail to provide a value from `net_version` calls that can be parsed into a decimal string. This PR makes our internally handling of this case more flexibly by using `null` as the network version value sent to the inpage provider when receiving an unexpected `net_version` call value.

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/27509?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/27487

## **Manual testing steps**

1. Open extension background
2. Intercept a request for net_version (you can use a proxy like Burpsuite) and make it return a bad value 
3. There should be no error in extension background related to this
4. Open a dapp
5. Open the dapp developer console
6. Expect to see an error like `JsonRpcError: MetaMask: Disconnected from chain. Attempting to connect.`
7. Test that `window.ethereum.request` works as expected despite this

## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
